### PR TITLE
`Development`: Fix flaky ExerciseGroupIntegrationTest

### PR DESCRIPTION
**Pull request by laadvo**
#### General
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

`importExerciseGroup_successfulWithExercisesIntoSameExam()` of `ExerciseGroupIntegrationTest` is flaky when running with postgres

### Description

Before, the test expected that the exercise groups are in a specific order in the list that gets returned by the POST request. In case they were not, the assertions failed. Now the test does not rely on the order anymore.

### Review Progress

You can test the test passing by running ` SPRING_PROFILES_INCLUDE=postgres ./gradlew test --tests ExerciseGroupIntegrationTest.importExerciseGroup_successfulWithExercisesIntoSameExam -x webapp` from your terminal and using `@RepeatedTest(X)` X being the number of repetitions.

#### Code Review
- [x] Code Review 1
- [ ] Code Review 2

### Test Coverage

unchanged

### CODE REVIEW

File: src/test/java/de/tum/in/www1/artemis/exam/ExerciseGroupIntegrationTest.java [hunk 1]
```diff
@@ -163,25 +164,20 @@ void testDeleteExerciseGroup_asEditor() throws Exception {
...
-                listExpected, ExerciseGroup.class, HttpStatus.OK);
-        assertThat(listReceived).hasSize(9);
+        final List<ExerciseGroup> exerciseGroupsNow = request.postListWithResponseBody(
+                "/api/courses/" + course1.getId() + "/exams/" + targetExam.getId() + "/import-exercise-group", exerciseGroupsBefore, ExerciseGroup.class, HttpStatus.OK);
 
-        // We import into the same exam -> double the exercise groups
-        listExpected.addAll(listExpected);
+        assertThat(exerciseGroupsNow).hasSize(9);
+        assertThat(exerciseGroupsNow).containsAll(exerciseGroupsBefore);
+        assertThat(exerciseGroupsNow).allMatch(element -> element.getId() != null);
```
Comment (Strohgelaender) [inline src/test/java/de/tum/in/www1/artemis/exam/ExerciseGroupIntegrationTest.java:174]: You can combine assertions of the same collection into one call.
```suggestion
        assertThat(exerciseGroupsNow).hasSize(9).containsAll(exerciseGroupsBefore).allMatch(element -> element.getId() != null);
```

File: src/test/java/de/tum/in/www1/artemis/exam/ExerciseGroupIntegrationTest.java [hunk 2]
```diff
@@ -163,25 +164,18 @@ void testDeleteExerciseGroup_asEditor() throws Exception {
...
-        for (int i = 5; i <= 8; i++) {
-            assertThat(listReceived.get(i).getId()).isNotNull();
-            assertThat(listReceived.get(i).getId()).isNotEqualTo(listExpected.get(i).getId());
-            assertThat(listReceived.get(i).getTitle()).isEqualTo(listExpected.get(i).getTitle());
-            assertThat(listReceived.get(i).getIsMandatory()).isEqualTo(listExpected.get(i).getIsMandatory());
+        for (var exerciseGroup : exerciseGroupsBefore) {
+            assertThat(exerciseGroupsNow).filteredOn(element -> Objects.equals(element.getId(), exerciseGroup.getId())).hasSize(1);
+
+            assertThat(exerciseGroupsNow).filteredOn(element -> Objects.equals(element.getTitle(), exerciseGroup.getTitle()))
+                    .filteredOn(element -> Objects.equals(element.getIsMandatory(), exerciseGroup.getIsMandatory())).hasSizeGreaterThan(1);
```
Comment (Strohgelaender) [inline src/test/java/de/tum/in/www1/artemis/exam/ExerciseGroupIntegrationTest.java:178]: Why greater than 1 here? Are there groups with duplicated titles?

Comment (laadvo) [reply src/test/java/de/tum/in/www1/artemis/exam/ExerciseGroupIntegrationTest.java:178]: As it is now, all groups seem to have the same title due to the ExamFactory helper function that is used. I think the test would make more sense if we change the helper function, which I will do now

**Review by Strohgelaender — APPROVED** [review]
makes sense, thanks for fixing the flakynes and for the test qualtify improvments.

**Review by krusche — APPROVED** [review]

**Pull request merged**